Fast Pair: publish Model IDs and update the documentation

alstrzebonski opened the pull request:
Updated documentation to publish Fast Pair demos' provisioning data.
Added default Fast Pair provisioning data that is used in the Fast Pair sample when no other provisioning data is specified.
Updated the documentation to align with the new way of displaying notifications for Fast Pair debug Model IDs.

Jira: NCSDK-20845
Jira: NCSDK-21329

alstrzebonski commented:
Please wait with the review till I update Battery Notification doc.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-11114/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-11114/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| desktop52_verification | X | | |
| test-fw-nrfconnect-ble | X | | |
| test-fw-nrfconnect-ble_samples | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-11114/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

alstrzebonski commented:
> Please wait with the review till I update Battery Notification doc. It will be better to review all updates together.

Battery Notification won't be a part of this PR as there are new problems with this extension and current documentation note remains valid.

Please review.

alstrzebonski commented:
Fixed conflicts and rebased on main.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-11114
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

annwoj commented on samples/bluetooth/peripheral_fast_pair/README.rst:
```diff
@@ -225,12 +236,21 @@ Test not discoverable advertising by completing `Testing`_ and the following add
+      :scale: 40 %
+      :alt: Fast Pair not discoverable advertising Android notification for debug Model ID
+
+   if you use debug Model ID.
```
```suggestion
```
Should be merged with the beginning of the sentence above

annwoj commented on samples/bluetooth/peripheral_fast_pair/README.rst:
```diff
@@ -225,12 +236,21 @@ Test not discoverable advertising by completing `Testing`_ and the following add
+      :alt: Fast Pair not discoverable advertising Android notification for debug Model ID
+
+   if you use debug Model ID.
+   The "Nordic" is replaced by your own Google account name as this is a default name written by the Fast Pair Seeker during initial pairing.
```
```suggestion
   The "Nordic" name is replaced by your own Google account name as this is a default name created by the Fast Pair Seeker during the initial pairing.
```

annwoj commented on samples/bluetooth/peripheral_fast_pair/README.rst:
```diff
@@ -173,13 +176,21 @@ Testing
+
+   If you use debug Model ID (e.g. default "NCS Fast Pair demo"), the notification is similar to the following one:
+
+   .. figure:: /images/bt_fast_pair_discoverable_notification_debug.png
```
@divipillai Should regular images have any specific frames?

divipillai replied:
yeah .png needs to be Width: 780 px and Height: 800 px as per the guidelines.

alstrzebonski replied:
Those are screenshots so the proportions (height to width ratio) depend really on what I want to show in the picture. Widening or lengthening the picture would look unnatural. Am I missing something?

annwoj replied:
In this case I think it's fine if we just make sure that the image parameters don't exceed the sizes mentioned by Divya. Would be great to get them closer to these parameters but without affecting the original shape of the image

mkapala-nordic commented on doc/nrf/external_comp/bt_fast_pair.rst:
```diff
@@ -48,6 +48,39 @@ Registering Fast Pair Provider
+1. NCS Fast Pair demo - general purpose Fast Pair provider. By default it's used by the :ref:`peripheral_fast_pair` sample. It has following information:
+
+   * Device name: NCS Fast Pair demo
+   * Model ID: 0x2A410B
```
Maybe it would be nice to enclose also model IDs inside backticks` ``<model_id>`` ` as suggested already for Anti-spoofing key?

alstrzebonski commented on doc/nrf/external_comp/bt_fast_pair.rst:
```diff
@@ -48,6 +48,39 @@ Registering Fast Pair Provider
 ==============================
 
 See the official `Fast Pair Model Registration`_ documentation for information how to register the device and obtain the Model ID and Anti-Spoofing Public/Private Key pair.
```
Sure

MarekPieta commented on doc/nrf/external_comp/bt_fast_pair.rst:
```diff
@@ -48,6 +48,39 @@ Registering Fast Pair Provider
 
 See the official `Fast Pair Model Registration`_ documentation for information how to register the device and obtain the Model ID and Anti-Spoofing Public/Private Key pair.
+Alternatively, you can use debug Model ID and Anti-Spoofing Public/Private Key pair obtained by Nordic for development purpose.
+Following debug devices have been registered:
```
Yes, I think we could move the provisioning data to nrf desktop documentation too (since we treat it as part of the application).

alstrzebonski commented on doc/nrf/external_comp/bt_fast_pair.rst:
```diff
@@ -48,6 +48,39 @@ Registering Fast Pair Provider
+   * Model ID: 0x2A410B
+   * Anti-spoofing key (base64, uncompressed): Unoh+nycK/ZJ7k3dHsdcNpiP1SfOy0P/Lx5XixyYois=
+   * Type: Input device
+   * Additional features: Data only connection, Support personalized name, Ringing device unsupported
```
It's a copy-paste from Google nearby devices console so I would rather keep the naming as it's on Google site.
![image](https://github.com/nrfconnect/sdk-nrf/assets/86837740/3ce07f97-c87e-45db-9af4-5de7714a9ed8)

annwoj replied:
Got it - thanks for the explanation :)

alstrzebonski commented on doc/nrf/external_comp/bt_fast_pair.rst:
```diff
@@ -48,6 +48,39 @@ Registering Fast Pair Provider
 
 See the official `Fast Pair Model Registration`_ documentation for information how to register the device and obtain the Model ID and Anti-Spoofing Public/Private Key pair.
+Alternatively, you can use debug Model ID and Anti-Spoofing Public/Private Key pair obtained by Nordic for development purpose.
+Following debug devices have been registered:
```
We can do that. Also following this comment: https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1190717455 the list will be split into according sample and desktop documentations.

alstrzebonski replied:
At second thought: I'd rather not do the table. Currently in nrf desktop it looks like that:
`* NCS keyboard - The Fast Pair Provider meant to be used with keyboards:

      * Device name: NCS keyboard
      * Model ID: ``0x52FF02``
      * Anti-spoofing key (base64, uncompressed): ``8E8ulwhSIp/skZeg27xmWv2SxRxTOagypHrf2OdrhGY=``
      * Type: Input device
      * Additional features: Data only connection, Support personalized name, Ringing device unsupported

   * NCS gaming mouse - Fast Pair Provider meant to be used with gaming mice:

      * Device name: NCS gaming mouse
      * Model ID: ``0x8E717D``
      * Anti-spoofing key (base64, uncompressed): ``dZxFzP7X9CcfLPC0apyRkmgsh3n2EbWo9NFNXfVuxAM=``
      * Type: Input device
      * Additional features: Data only connection, Support personalized name, Ringing device unsupported`

alstrzebonski replied:
Also additional features won't look good as they are long.

divipillai replied:
I think we can go with list option for now. @annwoj : Do you suggest any other option?

alstrzebonski commented on applications/nrf_desktop/README.rst:
```diff
@@ -863,9 +863,16 @@ Selecting a build type from command line
+   You can use your own provisioning data or provisioning data obtained by Nordic for development purpose.
+   Following debug devices, meant to be used with the nRF Desktop, have been registered:
+
+   * NCS keyboard - Fast Pair Provider meant to be used with keyboards.
+   * NCS gaming mouse - Fast Pair Provider meant to be used with gaming mouses.
```
@MarekPieta what do you think? I think it's correct.

kapi-no commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -451,6 +456,7 @@ Bluetooth libraries and services
 
     * Salt size in the Fast Pair not discoverable advertising from 1 byte to 2 bytes, to align with the Fast Pair specification update.
     * The :kconfig:option:`CONFIG_BT_FAST_PAIR_CRYPTO_OBERON` Kconfig option is now the default Fast Pair cryptographic backend.
+    * The Fast Pair documentation with debug Fast Pair provisioning data obtained for development purposes.
```
shouldn't we change this changelog entry as we no longer list the FP Models in the user guide?

You can find them now in the application and sample projects

kapi-no commented on samples/bluetooth/peripheral_fast_pair/README.rst:
```diff
@@ -57,10 +57,28 @@ See `Fast Pair Advertising`_ for detailed information about discoverable and not
-Before you can use a device as a Fast Pair Provider, you must register the device model with Google.
+Before you can use your device as a Fast Pair Provider, the device model must be registered with Google.
 This is required to obtain Model ID and Anti-Spoofing Private Key.
+You can register your own device or use the debug Model ID and Anti-Spoofing Public/Private Key pair obtained by Nordic for development purposes.
+By default, if Model ID and Anti-Spoofing Private Key are not specified, the following debug Fast Pair provider is used:
+
+* NCS Fast Pair demo - The input device Fast Pair provider:
+
+   * Device name: NCS Fast Pair demo
+   * Model ID: ``0x2A410B``
+   * Anti-spoofing key (base64, uncompressed): ``Unoh+nycK/ZJ7k3dHsdcNpiP1SfOy0P/Lx5XixyYois=``
+   * Type: Input device
+   * Additional features: Data only connection, Support personalized name, Ringing device unsupported
```
Can we use some kind of template doc mechanism for including this part of the documentation? We would need to have the possibility to provide some input arguments to it (e.g. Model ID = 0x2A410B, key, type, features)

The idea is to reuse the template in other places (e.g. nRF Desktop) and have a uniform way of phrasing this part of the documentation that is consistent across all projects.

@annwoj, @divipillai, is it possible?

divipillai replied:
yeah, I had the same idea. We can have include.txt for this content. If this cannot be fixed now, we can fix this in a separate PR.

alstrzebonski replied:
Good idea, I will work on it on monday, but if I don't manage to do it before release I will do it later.

annwoj commented on samples/bluetooth/peripheral_fast_pair/README.rst:
```diff
@@ -153,7 +171,8 @@ Building and running
 
-When building the sample, you must provide the Fast Pair Model ID (:c:macro:`FP_MODEL_ID`) and the Fast Pair Anti Spoofing Key (:c:macro:`FP_ANTI_SPOOFING_KEY`) as CMake options.
+When building the sample, you can provide the Fast Pair Model ID (:c:macro:`FP_MODEL_ID`) and the Fast Pair Anti-Spoofing Key (:c:macro:`FP_ANTI_SPOOFING_KEY`) as CMake options.
+If the data is not provided, the default provisioning data, obtained for the "NCS Fast Pair demo", the input device debug Fast Pair provider, is used.
```
```suggestion
If the data is not provided, the sample uses the default provisioning data obtained for the "NCS Fast Pair demo" (the input device debug Fast Pair provider).
```

@alstrzebonski Tired to rephrase this part a bit, please let me know in case it affects the meaning

alstrzebonski replied:
It's okay. Sounds better.

alstrzebonski commented:
Addressed all review comments beside this: https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1192461168 (it will be addressed later or in another PR).

divipillai commented on applications/nrf_desktop/README.rst:
```diff
@@ -863,11 +863,37 @@ Selecting a build type from command line
    * Registering a Fast Pair Provider
    * Provisioning a Fast Pair Provider in |NCS|
 
+.. important::
```
This important admonitions is used in two places. We could use the ".. include::" sytnax for this.
We could use the following labels before line no: 890 and after line no 896 respectively :

.. nrf_desktop_fastpair_important_start
.. nrf_desktop_fastpair_important_end

and in the sample documentation, where this admonitions is called, add the following "include" syntax.

.. include:: /applications/nrf_desktop/README.rst
   :start-after: nrf_desktop_fastpair_important_start
   :end-before: nrf_desktop_fastpair_important_end

divipillai replied:
This could be fixed along with the comment : https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1192461168, in that PR.

alstrzebonski replied:
Done. It needed a little reword because one was written in singular form and one was written in plural.

alstrzebonski commented:
Addressed all review comments beside this: https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1192461168 (it will be addressed later or in another PR).

alstrzebonski commented:
Rebased on main.

divipillai commented on applications/nrf_desktop/README.rst:
```diff
@@ -862,12 +862,40 @@ Selecting a build type from command line
    * Registering a Fast Pair Provider
    * Provisioning a Fast Pair Provider in |NCS|
 
+.. nrf_desktop_fastpair_important_start
+.. important::
```
```suggestion
.. nrf_desktop_fastpair_important_start

.. important::
```

divipillai commented on applications/nrf_desktop/README.rst:
```diff
@@ -862,12 +862,40 @@ Selecting a build type from command line
+   It should not be used in production.
+
+   To test with the debug mode Model ID, you must configure the Android device to include the debug results while displaying the nearby Fast Pair Providers.
+   For details, see `Verifying Fast Pair`_ in the GFPS documentation.
+.. nrf_desktop_fastpair_important_end
```
```suggestion
   For details, see `Verifying Fast Pair`_ in the GFPS documentation.
   
.. nrf_desktop_fastpair_important_end
```

alstrzebonski replied:
Addressed all review comments beside this: https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1192461168 (it will be addressed later or in another PR).

alstrzebonski commented:
Addressed all review comments beside this: https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1192461168 (it will be addressed later or in another PR).

alstrzebonski commented:
Rebased on main.

alstrzebonski commented:
https://github.com/nrfconnect/sdk-nrf/pull/11114#discussion_r1192461168 will be addressed in separate PR after the release.
